test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionBuilderExtensionsTests.cs

```diff
@@ -1,131 +1,131 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+//// Copyright (c) .NET Foundation. All rights reserved.
+//// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Linq;
-using System.Net;
-using System.Net.Http;
-using System.Security.Cryptography.X509Certificates;
-using Microsoft.AspNetCore.SignalR.Internal.Protocol;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Console;
-using Moq;
-using Xunit;
-using TransportType = Microsoft.AspNetCore.Http.Connections.TransportType;
+//using System;
+//using System.Linq;
+//using System.Net;
+//using System.Net.Http;
+//using System.Security.Cryptography.X509Certificates;
+//using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+//using Microsoft.Extensions.Logging;
+//using Microsoft.Extensions.Logging.Console;
+//using Moq;
+//using Xunit;
+//using TransportType = Microsoft.AspNetCore.Http.Connections.TransportType;
 
-namespace Microsoft.AspNetCore.SignalR.Client.Tests
-{
-    public class HubConnectionBuilderExtensionsTests
-    {
-        [Fact]
-        public void WithProxyRegistersGivenProxy()
-        {
-            var connectionBuilder = new HubConnectionBuilder();
-            var proxy = Mock.Of<IWebProxy>();
-            connectionBuilder.WithProxy(proxy);
-            Assert.Same(proxy, connectionBuilder.GetProxy());
```

Comment:
What level of testing is needed for getting and setting options on `WithHttpConnection`?

Because they're used to create `HttpConnection` there isn't a simple way to check they have been set. One idea is to make `_httpOptions` on `HttpConnection` internal.

Reply:
I don't think it's necessary to test individual setters/getters like this. Most things should be testable by creating the container, building the HubConnection and then testing some setting or behavior. If we need to expose internal options objects for testing as a last resort, we can do that, but I think it'd be preferable to test behaviors where possible since that's what the customer actually cares about.

Reply:
All the HTTP settings would really only be testable by starting the connection and sending a HTTP request. It starts to become less a simple unit test and more an integration test. Maybe having an IConnectionFactory will make it easier...

Reply:
> It starts to become less a simple unit test and more an integration test.

Yeah, I agree there. For some of these settings we already have integration tests that use these methods and I don't see a big reason to duplicate that logic here.